back-end/routes/users.js

```diff
@@ -1,55 +1,169 @@
     if (error) {
+      return res.status(500).send(error);
+    }
+    else if (!results || !results.rows || results.rows[0] === undefined) {
       return res.status(500).send('An error occurred');
     }
-    return res.status(200);
+    return res.status(200).send(results.rows);
   })
-}
+};
 
-/* GET all users */
-router.get('/', function(req, res, next) {
-  const getAllUsersQuery = 'SELECT * FROM users';
-  makeDbQueryAndReturnResults(getAllUsersQuery, res);
-});
+// async function getRowFromDb(queryString) {
+//   let results;
+//   try {
+//     results = await pool.query(queryString)
+//   } catch (e) {
+//     throw e
+//   }
+//   return results.rows[0];
+// };
 
+// TODO: Need to validate the user has access to this query
 /* GET user by ID */
 router.get('/:id', function(req, res, next) {
   const getUserQuery = 'SELECT * FROM users WHERE id=' + req.params.id;
   makeDbQueryAndReturnResults(getUserQuery, res);
 });
 
-/* POST to create new users */
-router.post('/', function(req, res, next) {
-  const {
-    username, password, github_username
-  } = req.body;
-
-  const createUserQuery = `INSERT INTO users(username, password, github_username, created_at) \
-VALUES('${username}', '${password}', '${github_username}', CURRENT_TIMESTAMP) returning *`
-  console.log(createUserQuery)
-  makeDbQueryAndReturnResults(createUserQuery, res);
+// UPDATED to be async for bcrypt library.
+router.post('/login', async function (req, res, next) {
+
+  // ie with a hashed password stored and checking against that.
+  try {
+    const query = "SELECT password, id FROM users WHERE email = ?";
+
+    const { email, password } = req.body;
+    let userID;
+    // initialize as invalid password value.
+    let passwordCheck = "bad";
+
+    // Not sure if this is the best way to do this query, maybe
+    // it should be a separate function.
+    pool.query(query, email, function(error, results, fields){
+      if(error){
+        res.write(JSON.stringify(error));
+        res.end();
+      }
+      // value changes if query worked.
+      passwordCheck = results[0];
+      userID = results[1];
+    });
+
+    if(await bcrypt.compare(password, passwordCheck)){
+      const token = jwt.sign({ userID }, process.env.ACCESS_TOKEN_SECRET);
+      res.json({
+        token: token
+      });
+    } else {
+      res.status(403).send('Login attempt failed.');
+    }
+  } catch(err) {
+    res.status(500).send(err);
+  }
+});
+
+// TODO: route for when logged in, what do we want to call this?
+// just using login/success as a placeholder for now,
+// can change if we want
+app.get('/login/success', ensureToken, function(req, res, next){
+  jwt.verify(req.token, process.env.ACCESS_TOKEN_SECRET, function(err, data){
+    if(err){
+      res.status(404).send('Bad token');
+    } else {
+      res.json({
+        text: 'successful login',
+        // TODO: send back relevant data, need to choose what that is
+        data: data
+      });
+    }
+  })
```

Comment:
We might want to encapsulate this in a "getIdFromToken" function that verifies the token and returns the user's id... this can then be used to validate access to the GET/:id and PUT/:id endpoints.